interactors/advanced-search.js

```diff
@@ -3,17 +3,23 @@ import Select from './select';
 import TextArea from './textarea';
 import HTML from './baseHTML';
 
+const getRowIndex = el => parseInt(el.getAttribute('data-row-index'), 10);
+
 export const AdvancedSearchRow = HTML.extend('advanced search row')
   .selector('[class*=AdvancedSearchRow-]')
   .filters({
-    index: (el) => parseInt(el.getAttribute('data-row-index'), 10),
-    query: (el) => el.querySelector('textarea').getAttribute('value'),
-    bool: (el) => el.querySelectorAll('select')[0].value,
-    option: (el) => el.querySelectorAll('select')[1].value,
+    index: el => getRowIndex(el),
+    query: el => el.querySelector('textarea').getAttribute('value'),
+    bool: el => el.querySelectorAll(`select#advanced-search-bool-${getRowIndex(el)}`).value,
+    matchOption: el => el.querySelectorAll(`select#advanced-search-match-${getRowIndex(el)}`).value,
```

Comment:
Here, the actual id of match option select would be `advanced-match-${rowIndex}`

Reply:
I changed the ids in the components to be consistent with the rest
https://github.com/folio-org/stripes-components/pull/2141/files#diff-1b9916b8e438211b16da2942c780ae6f074fd363ef3e4dc03b7f32cda2e9c8e2R121